OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -123,14 +123,17 @@ Actor ClosestProc(Actor self, Actor ignore)
 			var mi = self.Info.Traits.Get<MobileInfo>();
 			var path = self.World.WorldActor.Trait<IPathFinder>().FindPath(
 				PathSearch.FromPoints(self.World, mi, self, refs.Values.Select(r => r.Location), self.Location, false)
-					.WithCustomCost((loc) =>
+					.WithCustomCost(loc =>
 					{
 						if (!refs.ContainsKey(loc)) return 0;
 
 						var occupancy = refs[loc].Occupancy;
 
-						// 4 harvesters clogs up the refinery's delivery location:
-						if (occupancy >= 3) return int.MaxValue;
+						// 4 harvesters clogs up the refinery's delivery location,
+						// so if the refinery is crowded and there are other less crowded
+						// refineries, do not consider this one.
+						if (occupancy >= 3)
+							return int.MaxValue;
```

Comment:
We're back to excluding all refineries with 3 or more harvs even if there are no alternatives? Is that intentional? You should fix the commit message and the comments here if so because now they're just misleading.